lib/segment/src/index/field_index/binary_index.rs

```diff
@@ -268,15 +239,11 @@ impl PayloadFieldIndex for BinaryIndex {
             Some(Match::Value(MatchValue {
                 value: ValueVariants::Bool(value),
             })) => {
-                let iter = self
-                    .memory
-                    .iter()
-                    .zip(0u32..) // enumerate but with u32
-                    .filter_map(|(stored, point_id)| {
-                        stored.contains((*value).into()).then_some(point_id)
-                    });
-
-                Some(Box::new(iter))
+                if *value {
+                    Some(Box::new(self.memory.iter_has_true()))
+                } else {
+                    Some(Box::new(self.memory.iter_has_false()))
+                }
```

Comment:
We _can_ also do this here:

```suggestion
                Some(Box::new(if *value {
                    self.memory.iter_has_true()
                } else {
                    self.memory.iter_has_false()
                }))
```

Reply:
I see, thanks!

This _can_ probably be fixed by forcing it to be a generic iterator type, but that would over complicate it defeating the purpose.